transport/server.cc

```diff
@@ -239,14 +239,21 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
             ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
-                --_connections;
+            // Move conn to the background, announce, then block while monitoring for lifetime/errors.
+            (void)futurize_apply([this, conn] {
+                return advertise_new_connection(conn);
+            }).then_wrapped([this, conn] (future<> f) {
                 try {
                     f.get();
-                } catch (...) {
-                    clogger.debug("connection error: {}", std::current_exception());
+                } catch(...) {
+                    clogger.info("exception while advertising new connection: {}", std::current_exception());
                 }
+                return conn->process().finally([this, conn] {
+                    --_connections;
+                    return unadvertise_connection(conn);
+                }).handle_exception([] (std::exception_ptr ep) {
+                    clogger.info("exception found after connection close: {}", ep);
```

Comment:
Is "after connection close" the correct message for something that comes from conn->process()? Why "after connection close"? Note that before your patch the log message was about "connection error".
Also why did you change it from a debug message to an info? (maybe it makes sense, but I don't know why it was debug before).

Reply:
Ouch, absolutely, "close" is not the only thing that happens here. I updated exception description.
Also, I bumped loglevel because I think that connection errors are a severe thing and should not be buried in debug messages, as it used to be.